m4/dovecot.m4

```diff
@@ -342,6 +342,26 @@ AC_DEFUN([DC_DOVECOT_HARDENING],[
 	AC_LD_RELRO
 ])
 
+AC_DEFUN([DC_DOVECOT_FUZZER],[
+        AC_ARG_ENABLE(fuzzer,
+        AS_HELP_STRING([--enable-fuzzer=yes], [Enable fuzzer (default: no)]),
```

Comment:
should this be `--with-fuzzer` instead, like clang?

Reply:
I'm not really sure what the main difference is between "enable" and "with". I've been thinking that "with" is used when using some external library for example. This fuzzer can be just about compiler/linker options but then again it could be linking external library via `$LIB_FUZZING_ENGINE`. So from compiler flags point of view it's similar to `--enable-hardening=yes`. But then there's `--with-ioloop` and `--with-notify` which are about choosing which functions to use, even though they're not external libraries.. I wonder if those should have been using "enable" instead.

Anyway, I'm not sure. I could accept either way.

Reply:
btw. could you provide me with an email where we send the bug reports to? This is needed when merging to OSS-Fuzz (the email will be public in the oss-fuzz Repo)

Reply:
Here: oss-fuzz@open-xchange.com

Reply:
Perfect, I will fix up the OSS-Fuzz side once this PR is merged.

Reply:
Changed to use `--with-fuzzer` (And hopefully didn't change anything else, because I messed up a few times with git losing changes and had to restore them.)